lib/glimmer/tk/widget_proxy.rb

```diff
@@ -551,21 +562,91 @@ def handle_listener(listener_name, &listener)
       def on(listener_name, &listener)
         handle_listener(listener_name, &listener)
       end
-      
+
+      def raise_event(event_name, data = nil)
+        data = YAML.dump(data) if data && !data.is_a?(String)
+        tk.event_generate("<#{event_name}>", data: data)
+      end
+
       def unbind_all
         @listeners&.keys&.each do |key|
           if key.to_s.downcase.include?('command')
             @tk.send(key, '')
           else
-            @tk.bind(key, '')
+            @tk.bind_remove(key)
           end
         end
+        @listeners = nil
       end
-      
+
+      def clear
+        unbind_all
+        children.each(&:destroy)
+        @children = []
+      end
+
       def content(&block)
         Glimmer::DSL::Engine.add_content(self, Glimmer::DSL::Tk::WidgetExpression.new, keyword, *args, &block)
       end
 
+      def window?
+        false
+      end
+
+      def closest_window
+        widget = self
+        widget = widget.parent_proxy until widget.window?
+        widget
+      end
+
+      def close_window
+        closest_window.destroy
+      end
+
+      def visible
+        instance_variable_defined?(:@_visible) ? @_visible : (@_visible = true)
```

Comment:
I am not sure why you are using `@_visible` instead of `@visible`. The latter is preferred (`@visible`)

Also, this code could be simplified by avoiding an assignment that is wrapped by parentheses (and there is no need to use instance_variable_defined? usually because the instance variable will always be `nil` in that case):

```ruby
@visible = true if @visible.nil?
@visible
```

Reply:
Ah, my bad. Because of initially it was a Glimte patch for Glimmer, so I used "@_<variable>" in order not to interfere with any Glimmer variables. But since this is a PR into Glimmer itself, of course "_" is not needed anymore :)